WalletWasabi/Crypto/ZeroKnowledge/Prover.cs

```diff
@@ -0,0 +1,98 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public static class Prover
+	{
+		public static KnowledgeOfDiscreteLog CreateProof(Scalar secret, GroupElement publicPoint, GroupElement generator, WasabiRandom? random = null)
+		{
+			var proof = CreateProof(new[] { (secret, generator) }, publicPoint, random);
+
+			return new KnowledgeOfDiscreteLog(proof.Nonce, proof.Responses.First());
```

Comment:
what is the purpose of this separate type?

Reply:
we just wanted to take one step at a time. First Knowledge of DLog and then Knowledge of Representation. First, Schnorr protocol was written and made a lot of tests for that, and then we did not want to generalize it rather made a separate function. Strictly speaking we don't even need Knowledge of DLOG anymore, since Knowledge of Representation subsumes it.

Reply:
Actually it's not a separate type anymore. `KnowledgeOfDiscreteLog` inherits from `KnowledgeOfRepresentation`.